File: client/src/main/java/ir/asparsa/hobbytaste/ui/list/holder/UserNameViewHolder.java
package ir.asparsa.hobbytaste.ui.list.holder;

import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import ir.asparsa.android.ui.fragment.recycler.BaseRecyclerFragment;
import ir.asparsa.android.ui.list.holder.BaseViewHolder;
import ir.asparsa.hobbytaste.R;
import ir.asparsa.hobbytaste.ui.list.data.UsernameData;
import rx.functions.Action1;

/**
 * Created by hadi on 12/14/2016 AD.
 */
public class UserNameViewHolder extends BaseViewHolder<UsernameData> {

    @BindView(R.id.username_layout)
    ViewGroup mLayout;
    @BindView(R.id.username)
    TextView mUsernameTextView;

    public UserNameViewHolder(
            View itemView,
            Action1<BaseRecyclerFragment.Event> observer,
            Bundle savedInstanceState
    ) {
        super(itemView, observer, savedInstanceState);
        ButterKnife.bind(this, itemView);
    }

    @Override public void onBindView(final UsernameData data) {
        mUsernameTextView.setText(data.getUsername());
        mLayout.setOnClickListener(new View.OnClickListener() {
            @Override public void onClick(View view) {
                mObserver.call(new UsernameClick(data.getUsername()));
            }
        });
    }

    public static class UsernameClick implements BaseRecyclerFragment.Event {

        private String username;

        public UsernameClick(
                String username
        ) {
            this.username = username;
        }

        public String getUsername() {
            return username;
        }
    }

}

File: client/src/main/kotlin/ir/asparsa/hobbytaste/ui/list/holder/UserNameViewHolder.kt
package ir.asparsa.hobbytaste.ui.list.holder

import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import butterknife.ButterKnife
import com.github.kittinunf.reactiveandroid.view.rx_click
import ir.asparsa.android.core.logger.L
import ir.asparsa.android.ui.list.holder.BaseViewHolder
import ir.asparsa.hobbytaste.R
import ir.asparsa.hobbytaste.ui.list.data.UsernameData
import kotterknife.bindView
import rx.Observable

/**
 * @author hadi
 */
class UserNameViewHolder(
        itemView: View
) : BaseViewHolder<UsernameData>(itemView) {

    val mLayout: ViewGroup  by bindView(R.id.username_layout)
    val mUsernameTextView: TextView by bindView(R.id.username)

    init {
        ButterKnife.bind(this, itemView)
        L.d(javaClass, "On bind gets called")
    }

    override fun onBindView(data: UsernameData) {
        super.onBindView(data)
        L.d(javaClass, "On bind gets called")
        mUsernameTextView.text = data.username
    }

    fun clickStream(): Observable<UsernameData> {
        return mLayout.rx_click().map { data }
    }
}
